api/src/test/java/org/openmrs/CohortTest.java

```diff
@@ -165,4 +167,110 @@ public void setMemberIds_shouldSupportLargeCohorts() {
+	    membershipOne.setVoided(false);
+	    membershipOne.setEndDate(endDateLater);
+	    cohort.addMembership(membershipOne);
+	
+	    CohortMembership membershipTwo = new CohortMembership(8);
+	    membershipTwo.setVoided(true);
+	    cohort.addMembership(membershipTwo);
+	
+	    CohortMembership cohortMembershipThree = new CohortMembership(9);
+	    cohortMembershipThree.setEndDate(endDateEarlier);
+	    cohort.addMembership(cohortMembershipThree);
+	
+	    CohortMembership membershipFour = new CohortMembership(10);
+	    cohort.addMembership(membershipFour);
+	    
+	    assertTrue(cohort.hasActiveMembership(7));
+	    assertFalse(cohort.hasActiveMembership(9));
+	    assertFalse(cohort.hasActiveMembership(8));
+	    assertTrue(cohort.hasActiveMembership(10));
+	
+    }
+	
+	@Test
+	public void activeMembershipSize_shouldDetermineSizeCorrectly() throws Exception{
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date endDateEarlier = dateFormat.parse("2007-02-01 00:00:00");
+		Date endDateLater = dateFormat.parse("2100-02-01 00:00:00");
+		
+		
+		Cohort cohort = new Cohort(3);
+		CohortMembership temp = new CohortMembership(7);
+		temp.setVoided(false);
+		temp.setEndDate(endDateLater);
+		cohort.addMembership(temp);
+		
+		temp = new CohortMembership(8);
+		temp.setVoided(true);
+		cohort.addMembership(temp);
+		
+		temp = new CohortMembership(9);
+		temp.setEndDate(endDateEarlier);
+		cohort.addMembership(temp);
+		
+		temp = new CohortMembership(10);
+		temp.setVoided(false);
+		cohort.addMembership(temp);
+		
+		assertEquals(2, cohort.activeMembershipSize() );
+		
+	}
+	
+	
+	@Test
+	public void hasNoActiveMemberships_shouldReturnTrueIfNoneExists() throws Exception{
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date endDateEarlier = dateFormat.parse("2007-02-01 00:00:00");
+		Date endDateLater = dateFormat.parse("2100-02-01 00:00:00");
+		
+		
+		Cohort cohort = new Cohort(3);
+		CohortMembership temp = new CohortMembership(7);
+		temp.setVoided(true);
+		temp.setEndDate(endDateLater);
+		cohort.addMembership(temp);
+		
+		temp = new CohortMembership(8);
+		temp.setVoided(true);
+		cohort.addMembership(temp);
+		
+		temp = new CohortMembership(9);
+		temp.setEndDate(endDateEarlier);
+		cohort.addMembership(temp);
+		
+		temp = new CohortMembership(10);
+		temp.setVoided(true);
+		cohort.addMembership(temp);
+		
+		assertTrue(cohort.hasNoActiveMemberships());
+	}
+	
+	@Test
+	public void hasNoActiveMemberships_shouldReturnFalseIfOneExists() throws Exception{
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date endDateLater = dateFormat.parse("2100-02-01 00:00:00");
+		
+		
+		Cohort cohort = new Cohort(3);
+		CohortMembership temp = new CohortMembership(7);
+		temp.setVoided(false);
+		temp.setEndDate(endDateLater);
+		cohort.addMembership(temp);
+
+		
+		assertFalse(cohort.hasNoActiveMemberships());
```

Comment:
Will this test pass after the year 2100?

Reply:
Now it will since I used getInstance to get the date of today and then add one day. That way it will always. also when executed after 2100, work since the date is still in the future.